README.md

```diff
@@ -1,7 +1,298 @@
+* `filter` (⚠ does not exist yet) — returns a subset of scenarios, filtered using a callback;
+* `map` — lets you all scenarios in collection.
+
+Consider this example:
+
+```js
+// tests/scenarios.js
+import { Scenarios } from 'scenario-tester';
+import { Project } from 'fixturify-project';
+
+export default Scenarios
+
+  // Use a template project from filesystem
+  .fromProject(() => Project.fromDir(dirname(require.resolve('../test-app/package.json')), { linkDevDeps: true })))
+
+  // Define multiple scenarios, each scenario is derived from the template and has unique dependencies installed
+  .expand({
+    modern: async (project) => {
+      project.linkDevDependency('some-npm-package', { baseDir: __dirname, resolveName: 'some-npm-package' });
+      project.linkDevDependency('another-npm-package', { baseDir: __dirname, resolveName: 'another-npm-package' });
+    },
+
+    legacy: async function (project) => {
+      project.linkDevDependency('some-npm-package', { baseDir: __dirname, resolveName: 'some-npm-package--legacy' });
+      project.linkDevDependency('another-npm-package', { baseDir: __dirname, resolveName: 'another-npm-package--legacy' });
+    }
+  })
+```
+
+```js
+// tests/legacy-test.js
+import scenarios from './scenarios';
+import { module, test } from 'qunit';
+
+scenarios
+  // Pick one scenario
+  .only('legacy')
+
+  // Customize it
+  .map('without-another-npm-package', project => { // This callback can also be async
+    project.removeDevDependency('another-npm-package--legacy');
+  })
+
+  // Emit derivative projects to the filesystem
+  .forEachScenario(scenario => {
+
+    // Define your test suite
+    module(scenario.name, function (hooks) {
+      hooks.beforeEach(assert => {
+        // ...
+      });
+
+      test('output directories exist', async function () {
+        // ...
+      });
+
+      test('package.json is modified appropriately', async function () {
+        /// ...
+      });
+    });
+  });
+```
+
+## Command-line interface
+
+It is recommended that you define the following three commands in your `package.json` (the one specific to your scenarios).
+
+Example, customize to your needs:
+
+```json
+{
+  "scripts": {
+    "test": "qunit *-test.ts",
+    "test:list": "scenario-tester list",
+    "test:output": "scenario-tester output"
+  }
+}
+```
+
+The `test` command should be specific to your test suite of choice.
+
+The other two commands delegate to `scenario-tester`:
+
+* `scenario-tester list` — enumerates defined scenarios.
+
+    This command will output a list of all scenario names defined in your test suite.
+
+    Arguments:
+
+    * `--require <package-name>` — require a package before starting, e. g. `ts-node/register` if your test files are in TypeScript
+    * `--files <glob>` — which files to look in.
+    * `--matrix <format string>` — ❓ process scenario names with [util.format](https://nodejs.org/api/util.html#utilformatformat-args) and output as JSON.
```

Comment:
Maybe add a section about CI setup to the guide? Maybe take from https://github.com/ef4/scenario-tester/pull/10/files#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5R84

essentially `--matrix` is designed to communicate with GitHub and tell it what tests are available 👍

Reply:
I have mentioned that this flag is used in Github CI test matrix config, but have not provided full CI documentation. I suggest that it goes into `docs/` along with your project setup guide.